calyx-py/calyx/builder.py

```diff
@@ -465,6 +467,35 @@ def if_(
     return ast.If(port.expr, cg, as_control(body), as_control(else_body))
 
 
+class CellGroupBuilder:
+    """Just a cell and a group, for when it is convenient to pass them around together."""
+
+    def __init__(self, cell, group):
+        self.cell = cell
+        self.group = group
+
+
+def if_sugary(
```

Comment:
What do you think of the names `if_with` and `while_with`? We can make a more dramatic change and *require* that you build a `CellGroupBuilder` if you want to use comb groups and the normal `if_` and `while_` constructors will only take a port value.

Reply:
Done! Both the rename and the more dramatic change. I've piped the changes to all the other helpers (not just `eq`) and all the users of those helpers, so now everyone is using the sugary variants (since everyone so far _does_ need the combinational group structure. The old versions, plain ole `if_` and `while_`, can no longer accept comb groups.

Reply:
Thanks for taking a look. I have made a suggested change and also piped the basic idea of this PR all over the development. It is not ready for a re-review. The merge with the branch under it will be somewhat gruesome, so I'll request that you review this as it is, and then let me do the merge!